ACHTUNG! Compiler warnings from clang.

Warnings from clang. 
```
In file included from PicoDDS.cpp:50:
./PicoDDS.h:224:12: warning: 5 enumeration values not handled in switch: 'FORMAT_NONE', 'FORMAT_RGBA16', 'FORMAT_RGB16'... [-Wswitch]
                        switch( format)
                                ^
PicoDDS.cpp:100:8: warning: comparison of constant -1 with expression of type 'PicoDDS::ImgFormat' is always false [-Wtautological-constant-out-of-range-compare]
        if(-1 == imgdata_.format)
           ~~ ^  ~~~~~~~~~~~~~~~
2 warnings generated.
In file included from Drawables.cpp:4:
./Drawables.h:88:8: warning: private field 'm_width' is not used [-Wunused-private-field]
        float m_width;
              ^
1 warning generated.
BinaryConverter.cpp:26:42: warning: suggest braces around initialization of subobject [-Wmissing-braces]
const SGM_STRING_VALUE SGM_STRING_ID = { 's', 'g', 'm', SGM_VERSION };
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~
                                         {                         }
1 warning generated.
Loader.cpp:55:15: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
                        if (offset < 0 || offset > m_data->GetSize())
                            ~~~~~~ ^ ~
1 warning generated.
LuaModel.cpp:26:15: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
                if (pattern < 0 || pattern >= model->GetNumPatterns())
                    ~~~~~~~ ^ ~
1 warning generated.
In file included from Box.cpp:4:
./Box.h:50:9: warning: private field 'm_minAllocation' is not used [-Wunused-private-field]
        Uint32 m_minAllocation;
               ^
1 warning generated.
BaseSphere.cpp:114:12: warning: declaration shadows a local variable [-Wshadow]
                PosVert* vtxPtr = m_LatitudinalStrips[j]->Map<PosVert>(Graphics::BUFFER_MAP_WRITE);
                         ^
BaseSphere.cpp:83:11: note: previous declaration is here
        PosVert* vtxPtr = m_TriFanAbove->Map<PosVert>(Graphics::BUFFER_MAP_WRITE);
                 ^
1 warning generated.
In file included from GeoPatch.cpp:7:
./GeoPatchJobs.h:44:57: warning: function 'SBaseRequest' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
                patchID(0), edgeLen(0), fracStep(0.0), pTerrain(NULL) { assert(false); }
                                                                      ^
./GeoPatchJobs.h:78:66: warning: function 'SQuadSplitRequest' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
        SQuadSplitRequest(const SQuadSplitRequest &r) : SBaseRequest(r) { assert(false); }
                                                                        ^
./GeoPatchJobs.h:109:70: warning: function 'SSingleSplitRequest' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
        SSingleSplitRequest(const SSingleSplitRequest &r) : SBaseRequest(r)     { assert(false); }
                                                                                ^
In file included from GeoPatchJobs.cpp:5:
./GeoPatchJobs.h:44:57: warning: function 'SBaseRequest' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
                patchID(0), edgeLen(0), fracStep(0.0), pTerrain(NULL) { assert(false); }
                                                                      ^
./GeoPatchJobs.h:78:66: warning: function 'SQuadSplitRequest' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
        SQuadSplitRequest(const SQuadSplitRequest &r) : SBaseRequest(r) { assert(false); }
                                                                        ^
./GeoPatchJobs.h:109:70: warning: function 'SSingleSplitRequest' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
        SSingleSplitRequest(const SSingleSplitRequest &r) : SBaseRequest(r)     { assert(false); }
                                                                                ^
3 warnings generated.
3 warnings generated.
In file included from GeoSphere.cpp:8:
./GeoPatchJobs.h:44:57: warning: function 'SBaseRequest' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
                patchID(0), edgeLen(0), fracStep(0.0), pTerrain(NULL) { assert(false); }
                                                                      ^
./GeoPatchJobs.h:78:66: warning: function 'SQuadSplitRequest' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
        SQuadSplitRequest(const SQuadSplitRequest &r) : SBaseRequest(r) { assert(false); }
                                                                        ^
./GeoPatchJobs.h:109:70: warning: function 'SSingleSplitRequest' could be declared with attribute 'noreturn' [-Wmissing-noreturn]
        SSingleSplitRequest(const SSingleSplitRequest &r) : SBaseRequest(r)     { assert(false); }
                                                                                ^
3 warnings generated.
KeyBindings.cpp:437:18: warning: comparison of constant -1 with expression of type 'Uint8' (aka 'unsigned char') is always false [-Wtautological-constant-out-of-range-compare]
        if (ab.joystick == -1) {
            ~~~~~~~~~~~ ^  ~~
1 warning generated.
In file included from SectorView.cpp:12:
./SectorView.h:96:8: warning: private field 'm_game' is not used [-Wunused-private-field]
        Game* m_game;
              ^
./SectorView.h:125:20: warning: private field 'm_galaxyButton' is not used [-Wunused-private-field]
        Gui::ImageButton *m_galaxyButton;
                          ^
2 warnings generated.
StringF.cpp:363:27: warning: unused function 'scan_unformatted' [-Wunused-function]
static inline const char* scan_unformatted(const char* fmt) {
                          ^
1 warning generated.
Ship.cpp:1249:60: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                const float speed = std::min(10.0*GetVelocity().Length()*abs(m_thrusters.z),100.0);
                                                                         ^
Ship.cpp:1249:60: note: use function 'std::abs' instead
                const float speed = std::min(10.0*GetVelocity().Length()*abs(m_thrusters.z),100.0);
                                                                         ^~~
                                                                         std::abs
Ship.cpp:34:21: warning: unused variable 'KINETIC_ENERGY_MULT' [-Wunused-const-variable]
static const double KINETIC_ENERGY_MULT = 0.01;
                    ^
ShipCockpit.cpp:70:32: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
        float gforce = Clamp(floorf(((abs(cur_vel) - m_shipVel) / timeStep) / 9.8f), -COCKPIT_MAX_GFORCE, COCKPIT_MAX_GFORCE);
                                      ^
ShipCockpit.cpp:70:32: note: use function 'std::abs' instead
        float gforce = Clamp(floorf(((abs(cur_vel) - m_shipVel) / timeStep) / 9.8f), -COCKPIT_MAX_GFORCE, COCKPIT_MAX_GFORCE);
                                      ^~~
                                      std::abs
ShipCockpit.cpp:71:6: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
        if (abs(cur_vel) > 500000.0f ||      // Limit gforce measurement so we don't get astronomical fluctuations
            ^
ShipCockpit.cpp:71:6: note: use function 'std::abs' instead
        if (abs(cur_vel) > 500000.0f ||      // Limit gforce measurement so we don't get astronomical fluctuations
            ^~~
            std::abs
ShipCockpit.cpp:72:3: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
                abs(gforce - m_gForce) > 100.0) { // Smooth out gforce one frame spikes, sometimes happens when hitting max speed due to the thrust limiters
                ^
ShipCockpit.cpp:72:3: note: use function 'std::abs' instead
                abs(gforce - m_gForce) > 100.0) { // Smooth out gforce one frame spikes, sometimes happens when hitting max speed due to the thrust limiters
                ^~~
                std::abs
ShipCockpit.cpp:75:6: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
        if (abs(gforce - m_gForce) > 100.0) {
            ^
ShipCockpit.cpp:75:6: note: use function 'std::abs' instead
        if (abs(gforce - m_gForce) > 100.0) {
            ^~~
            std::abs
ShipCockpit.cpp:78:6: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
        if (abs(m_translate.z - m_offset) < 0.001f) {
            ^
ShipCockpit.cpp:78:6: note: use function 'std::abs' instead
        if (abs(m_translate.z - m_offset) < 0.001f) {
            ^~~
            std::abs
5 warnings generated.
2 warnings generated.
In file included from SystemInfoView.cpp:8:
./SystemInfoView.h:76:25: warning: private field 'm_solidState' is not used [-Wunused-private-field]
        Graphics::RenderState *m_solidState;
                               ^
SystemView.cpp:728:22: warning: unused variable 'fogDensity' [-Wunused-const-variable]
static const GLfloat fogDensity = 0.1f;
                     ^
SystemView.cpp:729:22: warning: unused variable 'fogColor' [-Wunused-const-variable]
static const GLfloat fogColor[4] = { 0,0,0,1.0f };
                     ^
1 warning generated.
2 warnings generated.
WorldView.cpp:2146:15: warning: declaration shadows a variable in namespace '(anonymous)' [-Wshadow]
                const Color green = Color(0, 255, 0, 204);
                            ^
WorldView.cpp:52:21: note: previous declaration is here
        static const Color green(0, 255, 0, 204);
                           ^
1 warning generated.
```

--- a/contrib/PicoDDS/PicoDDS.cpp
+++ b/contrib/PicoDDS/PicoDDS.cpp
@@ -76,7 +76,7 @@ size_t DDSImage::Read(const char* pData, const size_t dataSize)
 {
 	// Read in header and decode
 	if (!ReadHeader(pData, surfacedata_))
-		return -1;
+		return 0;
 
 	if (surfacedata_.mipmapcount==0)
 		surfacedata_.mipmapcount=1;
@@ -95,10 +95,10 @@ size_t DDSImage::Read(const char* pData, const size_t dataSize)
 	imgdata_.numImages = GetNumImages();
 	imgdata_.size = CalculateStoreageSize();
 	if(0 >= imgdata_.size)
-		return -1;
+		return 0;
 			
-	if(-1 == imgdata_.format)
-		return -1;
+	if(FORMAT_NONE == imgdata_.format)
+		return 0;
 
 	const long headerSize=128;
 	const size_t DDSStructSize = sizeof(DDS::DDSStruct)+4;
@@ -321,7 +321,7 @@ ImgFormat DDSImage::GetDXTFormat() const
 #ifdef PICODDS_OPENGL
 int DDSImage::GetOpenGLFormat()
 {
-	int format = -1;
+	int format = 0;
 	switch(surfacedata_.pixelformat.fourCC) 
 	{
 	case FOURCC('D','X','T','1'):

--- a/contrib/PicoDDS/PicoDDS.h
+++ b/contrib/PicoDDS/PicoDDS.h
@@ -234,11 +234,15 @@ namespace PicoDDS
 			case FORMAT_A8L8:
 			case FORMAT_L16:
 			case FORMAT_V8U8:
+			case FORMAT_A16:
 				return numPixels*2;
 
 			case FORMAT_RGB:
 			case FORMAT_BGR:
 				return numPixels*3;
+
+			case FORMAT_RGB16:
+				return numPixels*6;
 
 			case FORMAT_RGBA:
 			case FORMAT_BGRA:
@@ -248,8 +252,10 @@ namespace PicoDDS
 			case FORMAT_V16U16:
 			case FORMAT_G16R16:
 			case FORMAT_Q8W8V8U8:
+			case FORMAT_A16L16:
 				return numPixels*4;
 
+			case FORMAT_RGBA16:
 			case FORMAT_R16G16B16A16F:
 			case FORMAT_G32R32F:
 				return numPixels*8;
@@ -265,8 +271,11 @@ namespace PicoDDS
 			case FORMAT_DXT5:
 			case FORMAT_3DC:
 				return ((width+3)/4) * ((height+3)/4) * depth * 16;
+
+			case FORMAT_NONE:
+				return 0;
 			}
-			return -1;
+			return 0;
 		}
 
 		int CalculateStoreageSize() const;

--- a/src/BaseSphere.cpp
+++ b/src/BaseSphere.cpp
@@ -111,7 +111,7 @@ void BaseSphere::DrawAtmosphereSurface(Graphics::Renderer *renderer,
 		const float sinLat = sin(lat);
 		const float cosLat2 = cos(lat+latDiff);
 		const float sinLat2 = sin(lat+latDiff);
-		PosVert* vtxPtr = m_LatitudinalStrips[j]->Map<PosVert>(Graphics::BUFFER_MAP_WRITE);
+		vtxPtr = m_LatitudinalStrips[j]->Map<PosVert>(Graphics::BUFFER_MAP_WRITE);
 		vtxPtr[0].pos = vector3f(0.f, 1.f, 0.f);
 		for (int i=0; i<=LONG_SEGS; i++) {
 			vtxPtr[(i*2)+0].pos = vector3f(sinLat*sinTable[i], cosLat, -sinLat*cosTable[i]);

--- a/src/GeoPatchJobs.h
+++ b/src/GeoPatchJobs.h
@@ -40,8 +40,7 @@ public:
 
 protected:
 	// deliberately prevent copy constructor access
-	SBaseRequest(const SBaseRequest &r) : v0(0.0), v1(0.0), v2(0.0), v3(0.0), centroid(0.0), depth(0), 
-		patchID(0), edgeLen(0), fracStep(0.0), pTerrain(NULL) { assert(false); }
+	SBaseRequest(const SBaseRequest &r) = delete;
 };
 
 class SQuadSplitRequest : public SBaseRequest {
@@ -75,7 +74,7 @@ public:
 
 protected:
 	// deliberately prevent copy constructor access
-	SQuadSplitRequest(const SQuadSplitRequest &r) : SBaseRequest(r)	{ assert(false); }
+	SQuadSplitRequest(const SQuadSplitRequest &r) = delete;
 };
 
 class SSingleSplitRequest : public SBaseRequest {
@@ -106,7 +105,7 @@ public:
 
 protected:
 	// deliberately prevent copy constructor access
-	SSingleSplitRequest(const SSingleSplitRequest &r) : SBaseRequest(r)	{ assert(false); }
+	SSingleSplitRequest(const SSingleSplitRequest &r) = delete;
 };
 
 class SBaseSplitResult {

--- a/src/KeyBindings.cpp
+++ b/src/KeyBindings.cpp
@@ -433,10 +433,13 @@ bool AxisBinding::FromString(const char *str, AxisBinding &ab) {
 	// force terminate
 	joyUUIDBuf[JoyUUIDLength-1] = '\0';
 	// now, map the GUID to a joystick number
-	ab.joystick = Pi::JoystickFromGUIDString(joyUUIDBuf);
-	if (ab.joystick == -1) {
+	const int joystick = Pi::JoystickFromGUIDString(joyUUIDBuf);
+	if (joystick == -1) {
 		return false;
 	}
+	// found a joystick
+	assert(joystick < 256);
+	ab.joystick = Uint8(joystick);
 
 	if (strncmp(p, "Axis", 4) != 0)
 		return false;

--- a/src/SectorView.cpp
+++ b/src/SectorView.cpp
@@ -45,7 +45,7 @@ enum DetailSelection {
 static const float ZOOM_SPEED = 15;
 static const float WHEEL_SENSITIVITY = .03f;		// Should be a variable in user settings.
 
-SectorView::SectorView(Game* game) : UIView(), m_game(game), m_galaxy(game->GetGalaxy())
+SectorView::SectorView(Game* game) : UIView(), m_galaxy(game->GetGalaxy())
 {
 	InitDefaults();
 
@@ -76,7 +76,7 @@ SectorView::SectorView(Game* game) : UIView(), m_game(game), m_galaxy(game->GetG
 	InitObject();
 }
 
-SectorView::SectorView(const Json::Value &jsonObj, Game* game) : UIView(), m_game(game), m_galaxy(game->GetGalaxy())
+SectorView::SectorView(const Json::Value &jsonObj, Game* game) : UIView(), m_galaxy(game->GetGalaxy())
 {
 	InitDefaults();
 

--- a/src/SectorView.h
+++ b/src/SectorView.h
@@ -93,7 +93,6 @@ private:
 	void OnKeyPressed(SDL_Keysym *keysym);
 	void OnSearchBoxKeyPress(const SDL_Keysym *keysym);
 
-	Game* m_game;
 	RefCountedPtr<Galaxy> m_galaxy;
 
 	bool m_inSystem;
@@ -122,7 +121,6 @@ private:
 	Gui::Label *m_zoomLevelLabel;
 	Gui::ImageButton *m_zoomInButton;
 	Gui::ImageButton *m_zoomOutButton;
-	Gui::ImageButton *m_galaxyButton;
 	Gui::TextEntry *m_searchBox;
 	Gui::Label *m_statusLabel;
 	Gui::ToggleButton *m_drawOutRangeLabelButton;

--- a/src/Ship.cpp
+++ b/src/Ship.cpp
@@ -1245,7 +1245,7 @@ void Ship::StaticUpdate(const float timeStep)
 	//Add smoke trails for missiles on thruster state
 	if (m_type->tag == ShipType::TAG_MISSILE && m_thrusters.z < 0.0 && 0.1*Pi::rng.Double() < timeStep) {
 		const vector3d pos = GetOrient() * vector3d(0, 0 , 5);
-		const float speed = std::min(10.0*GetVelocity().Length()*std::abs(m_thrusters.z),100.0);
+		const float speed = std::min(10.0*GetVelocity().Length()*fabs(m_thrusters.z),100.0);
 		Sfx::AddThrustSmoke(this, Sfx::TYPE_SMOKE, speed, pos);
 	}
 }

--- a/src/Ship.cpp
+++ b/src/Ship.cpp
@@ -31,7 +31,6 @@
 #include <algorithm>
 
 static const float TONS_HULL_PER_SHIELD = 10.f;
-static const double KINETIC_ENERGY_MULT	= 0.01;
 HeatGradientParameters_t Ship::s_heatGradientParams;
 const float Ship::DEFAULT_SHIELD_COOLDOWN_TIME = 1.0f;
 
@@ -1246,7 +1245,7 @@ void Ship::StaticUpdate(const float timeStep)
 	//Add smoke trails for missiles on thruster state
 	if (m_type->tag == ShipType::TAG_MISSILE && m_thrusters.z < 0.0 && 0.1*Pi::rng.Double() < timeStep) {
 		const vector3d pos = GetOrient() * vector3d(0, 0 , 5);
-		const float speed = std::min(10.0*GetVelocity().Length()*abs(m_thrusters.z),100.0);
+		const float speed = std::min(10.0*GetVelocity().Length()*std::abs(m_thrusters.z),100.0);
 		Sfx::AddThrustSmoke(this, Sfx::TYPE_SMOKE, speed, pos);
 	}
 }

--- a/src/ShipCockpit.cpp
+++ b/src/ShipCockpit.cpp
@@ -67,15 +67,15 @@ void ShipCockpit::Update(float timeStep)
 
 	//---------------------------------------- Acceleration
 	float cur_vel = CalculateSignedForwardVelocity(-cur_dir, Pi::player->GetVelocity()); // Forward is -Z
-	float gforce = Clamp(floorf(((abs(cur_vel) - m_shipVel) / timeStep) / 9.8f), -COCKPIT_MAX_GFORCE, COCKPIT_MAX_GFORCE);
-	if (abs(cur_vel) > 500000.0f ||      // Limit gforce measurement so we don't get astronomical fluctuations
-		abs(gforce - m_gForce) > 100.0) { // Smooth out gforce one frame spikes, sometimes happens when hitting max speed due to the thrust limiters
+	float gforce = Clamp(floorf(((std::abs(cur_vel) - m_shipVel) / timeStep) / 9.8f), -COCKPIT_MAX_GFORCE, COCKPIT_MAX_GFORCE);
+	if (std::abs(cur_vel) > 500000.0f ||      // Limit gforce measurement so we don't get astronomical fluctuations
+		std::abs(gforce - m_gForce) > 100.0) { // Smooth out gforce one frame spikes, sometimes happens when hitting max speed due to the thrust limiters
 		gforce = 0.0f;
 	}
-	if (abs(gforce - m_gForce) > 100.0) {
+	if (std::abs(gforce - m_gForce) > 100.0) {
 		gforce = 0.0f;
 	}
-	if (abs(m_translate.z - m_offset) < 0.001f) {
+	if (std::abs(m_translate.z - m_offset) < 0.001f) {
 		m_transInterp = 0.0f;
 	}
 	float offset = (gforce > 14.0f ? -1.0f : (gforce < -14.0f ? 1.0f : 0.0f)) * COCKPIT_ACCEL_OFFSET;

--- a/src/ShipCockpit.cpp
+++ b/src/ShipCockpit.cpp
@@ -67,15 +67,15 @@ void ShipCockpit::Update(float timeStep)
 
 	//---------------------------------------- Acceleration
 	float cur_vel = CalculateSignedForwardVelocity(-cur_dir, Pi::player->GetVelocity()); // Forward is -Z
-	float gforce = Clamp(floorf(((std::abs(cur_vel) - m_shipVel) / timeStep) / 9.8f), -COCKPIT_MAX_GFORCE, COCKPIT_MAX_GFORCE);
-	if (std::abs(cur_vel) > 500000.0f ||      // Limit gforce measurement so we don't get astronomical fluctuations
-		std::abs(gforce - m_gForce) > 100.0) { // Smooth out gforce one frame spikes, sometimes happens when hitting max speed due to the thrust limiters
+	float gforce = Clamp(floorf(((fabs(cur_vel) - m_shipVel) / timeStep) / 9.8f), -COCKPIT_MAX_GFORCE, COCKPIT_MAX_GFORCE);
+	if (fabs(cur_vel) > 500000.0f ||      // Limit gforce measurement so we don't get astronomical fluctuations
+		fabs(gforce - m_gForce) > 100.0) { // Smooth out gforce one frame spikes, sometimes happens when hitting max speed due to the thrust limiters
 		gforce = 0.0f;
 	}
-	if (std::abs(gforce - m_gForce) > 100.0) {
+	if (fabs(gforce - m_gForce) > 100.0) {
 		gforce = 0.0f;
 	}
-	if (std::abs(m_translate.z - m_offset) < 0.001f) {
+	if (fabs(m_translate.z - m_offset) < 0.001f) {
 		m_transInterp = 0.0f;
 	}
 	float offset = (gforce > 14.0f ? -1.0f : (gforce < -14.0f ? 1.0f : 0.0f)) * COCKPIT_ACCEL_OFFSET;

--- a/src/SystemInfoView.h
+++ b/src/SystemInfoView.h
@@ -72,8 +72,6 @@ private:
 	std::vector<std::pair<Uint32, BodyIcon*> > m_bodyIcons;
 	bool m_unexplored;
 	bool m_hasTradeAnalyzer;
-
-	Graphics::RenderState *m_solidState;
 };
 
 #endif /* _SYSTEMINFOVIEW_H */

--- a/src/SystemView.cpp
+++ b/src/SystemView.cpp
@@ -96,7 +96,7 @@ static std::string formatTime(double t)
 {
 	std::stringstream formattedTime;
 	formattedTime << std::setprecision(1) << std::fixed;
-	double absT = std::abs(t);
+	double absT = fabs(t);
 	if(absT < 60.)
 		formattedTime << t << "s";
 	else if(absT < 3600)

--- a/src/SystemView.cpp
+++ b/src/SystemView.cpp
@@ -725,9 +725,6 @@ void SystemView::PutSelectionBox(const vector3d &worldPos, const Color &col)
 	Gui::Screen::LeaveOrtho();
 }
 
-static const GLfloat fogDensity = 0.1f;
-static const GLfloat fogColor[4] = { 0,0,0,1.0f };
-
 void SystemView::GetTransformTo(const SystemBody *b, vector3d &pos)
 {
 	if (b->GetParent()) {

--- a/src/WorldView.cpp
+++ b/src/WorldView.cpp
@@ -2143,8 +2143,6 @@ void NavTunnelWidget::Draw() {
 		const vector3d eyevec = rotmat * m_worldView->m_activeCameraController->GetOrient().VectorZ();
 		if (eyevec.Dot(navpos) >= 0.0) return;
 
-		const Color green = Color(0, 255, 0, 204);
-
 		const double distToDest = Pi::player->GetPositionRelTo(navtarget).Length();
 
 		const int maxSquareHeight = std::max(Gui::Screen::GetWidth(), Gui::Screen::GetHeight()) / 2;

--- a/src/graphics/Drawables.cpp
+++ b/src/graphics/Drawables.cpp
@@ -215,7 +215,7 @@ void Line3D::Dirty() {
 }
 //------------------------------------------------------------
 
-Lines::Lines() : m_refreshVertexBuffer(true), m_width(2.0f), m_va(new VertexArray(Graphics::ATTRIB_POSITION | Graphics::ATTRIB_DIFFUSE))
+Lines::Lines() : m_refreshVertexBuffer(true), m_va(new VertexArray(Graphics::ATTRIB_POSITION | Graphics::ATTRIB_DIFFUSE))
 {
 	PROFILE_SCOPED()
 	// XXX bug in Radeon drivers will cause crash in glLineWidth if width >= 3

--- a/src/graphics/Drawables.h
+++ b/src/graphics/Drawables.h
@@ -85,7 +85,6 @@ private:
 	void CreateVertexBuffer(Graphics::Renderer *r, const Uint32 size);
 
 	bool m_refreshVertexBuffer;
-	float m_width;
 	RefCountedPtr<Material> m_material;
 	RefCountedPtr<VertexBuffer> m_vertexBuffer;
 	std::unique_ptr<VertexArray> m_va;

--- a/src/scenegraph/BinaryConverter.cpp
+++ b/src/scenegraph/BinaryConverter.cpp
@@ -23,7 +23,7 @@ union SGM_STRING_VALUE{
 	char name[4];
 	Uint32 value;
 };
-const SGM_STRING_VALUE SGM_STRING_ID = { 's', 'g', 'm', SGM_VERSION };
+const SGM_STRING_VALUE SGM_STRING_ID = { {'s', 'g', 'm', SGM_VERSION} };
 const std::string SGM_EXTENSION = ".sgm";
 const std::string SAVE_TARGET_DIR = "binarymodels";
 

--- a/src/scenegraph/Loader.cpp
+++ b/src/scenegraph/Loader.cpp
@@ -52,7 +52,7 @@ namespace {
 				case aiOrigin_END: offset += m_data->GetSize(); break;
 				default: assert(0); break;
 			}
-			if (offset < 0 || offset > m_data->GetSize())
+			if (offset > m_data->GetSize())
 				return aiReturn_FAILURE;
 			m_cursor = m_data->GetData() + offset;
 			return aiReturn_SUCCESS;

--- a/src/scenegraph/LuaModel.cpp
+++ b/src/scenegraph/LuaModel.cpp
@@ -23,7 +23,7 @@ public:
 			return luaL_error(l, "model '%s' does not support patterns", model->GetName().c_str());
 		unsigned int pattern = luaL_checkinteger(l, 2);
 		pattern--; // Lua counts from 1
-		if (pattern < 0 || pattern >= model->GetNumPatterns())
+		if (pattern >= model->GetNumPatterns())
 			return luaL_error(l, "invalid pattern number '%d'; model '%s' has patterns 1-%d", pattern+1, model->GetName().c_str(), model->GetNumPatterns());
 		model->SetPattern(pattern);
 		return 0;

--- a/src/ui/Box.h
+++ b/src/ui/Box.h
@@ -47,7 +47,6 @@ private:
 	std::list<Child> m_children;
 
 	Point m_preferredSize;
-	Uint32 m_minAllocation;
 	Uint32 m_numVariable;
 };
 

--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -23,7 +23,7 @@ std::string format_money(double cents, bool showCents){
 
 	const char *format = (money < 0) ? "-$%.2f" : "$%.2f";
 	char buf[64];
-	snprintf(buf, sizeof(buf), format, std::abs(money));
+	snprintf(buf, sizeof(buf), format, fabs(money));
 	std::string result(buf);
 
 	size_t pos = result.find_first_of('.');      // pos to decimal point
@@ -36,7 +36,7 @@ std::string format_money(double cents, bool showCents){
 	size_t groupMin = strtol(Lang::NUMBER_GROUP_MIN, &end, 10);
 	assert(*end == 0);
 
-	if(groupDigits != 0 && std::abs(money) >= groupMin){
+	if(groupDigits != 0 && fabs(money) >= groupMin){
 
 		std::string groupSep = std::string(Lang::NUMBER_GROUP_SEP) == " " ?
 			"\u00a0" : Lang::NUMBER_GROUP_SEP;     // space should be fixed space

